Fix warnings in the code.

**Describe the bug**
<details>
    <summary>Current warnings that show up</summary>

```
../components/Util/src/Base64.cpp:29:13: warning: 'void util::a4_to_a3(uint8_t*, uint8_t*)' defined but not used [-Wunused-function]
 static void a4_to_a3(std::uint8_t* a3, std::uint8_t* a4)
             ^~~~~~~~
[1171/1199] Building CXX object esp-idf/Network/CMakeFiles/__idf_Network.dir/src/WiFi.cpp.obj
In file included from ../components/Network/src/WiFiEventHandler.h:18,
                 from ../components/Network/src/WiFi.h:22,
                 from ../components/Network/src/WiFi.cpp:12:
/home/jeandudey/Dev/esp-idf/components/esp_event/include/esp_event_loop.h:2:2: warning: #warning "esp_event_loop.h is deprecated, please include esp_event.h instead" [-Wcpp]
 #warning "esp_event_loop.h is deprecated, please include esp_event.h instead"
  ^~~~~~~
../components/Network/src/WiFi.cpp: In member function 'esp_err_t network::WiFi::init()':
../components/Network/src/WiFi.cpp:113:24: warning: 'void tcpip_adapter_init()' is deprecated [-Wdeprecated-declarations]
     tcpip_adapter_init();
                        ^
In file included from /home/jeandudey/Dev/esp-idf/components/esp_netif/include/esp_netif.h:35,
                 from /home/jeandudey/Dev/esp-idf/components/esp_event/include/esp_event_legacy.h:22,
                 from /home/jeandudey/Dev/esp-idf/components/esp_event/include/esp_event.h:26,
                 from ../components/Network/src/WiFi.h:18,
                 from ../components/Network/src/WiFi.cpp:12:
/home/jeandudey/Dev/esp-idf/components/tcpip_adapter/include/tcpip_adapter.h:34:6: note: declared here
 void tcpip_adapter_init(void)  __attribute__ ((deprecated));
      ^~~~~~~~~~~~~~~~~~
../components/Network/src/WiFi.cpp:113:24: warning: 'void tcpip_adapter_init()' is deprecated [-Wdeprecated-declarations]
     tcpip_adapter_init();
                        ^
In file included from /home/jeandudey/Dev/esp-idf/components/esp_netif/include/esp_netif.h:35,
                 from /home/jeandudey/Dev/esp-idf/components/esp_event/include/esp_event_legacy.h:22,
                 from /home/jeandudey/Dev/esp-idf/components/esp_event/include/esp_event.h:26,
                 from ../components/Network/src/WiFi.h:18,
                 from ../components/Network/src/WiFi.cpp:12:
/home/jeandudey/Dev/esp-idf/components/tcpip_adapter/include/tcpip_adapter.h:34:6: note: declared here
 void tcpip_adapter_init(void)  __attribute__ ((deprecated));
      ^~~~~~~~~~~~~~~~~~
../components/Network/src/WiFi.cpp:117:50: warning: 'esp_err_t esp_event_loop_init(system_event_cb_t, void*)' is deprecated [-Wdeprecated-declarations]
     esp_event_loop_init(&WiFi::eventHandler, this);
                                                  ^
In file included from /home/jeandudey/Dev/esp-idf/components/esp_event/include/esp_event.h:26,
                 from ../components/Network/src/WiFi.h:18,
                 from ../components/Network/src/WiFi.cpp:12:
/home/jeandudey/Dev/esp-idf/components/esp_event/include/esp_event_legacy.h:227:11: note: declared here
 esp_err_t esp_event_loop_init(system_event_cb_t cb, void *ctx) __attribute__ ((deprecated));
           ^~~~~~~~~~~~~~~~~~~
../components/Network/src/WiFi.cpp:117:50: warning: 'esp_err_t esp_event_loop_init(system_event_cb_t, void*)' is deprecated [-Wdeprecated-declarations]
     esp_event_loop_init(&WiFi::eventHandler, this);
                                                  ^
In file included from /home/jeandudey/Dev/esp-idf/components/esp_event/include/esp_event.h:26,
                 from ../components/Network/src/WiFi.h:18,
                 from ../components/Network/src/WiFi.cpp:12:
/home/jeandudey/Dev/esp-idf/components/esp_event/include/esp_event_legacy.h:227:11: note: declared here
 esp_err_t esp_event_loop_init(system_event_cb_t cb, void *ctx) __attribute__ ((deprecated));
           ^~~~~~~~~~~~~~~~~~~
[1173/1199] Building CXX object esp-idf/Network/CMakeFiles/__idf_Network.dir/src/WiFiEventHandler.cpp.obj
In file included from ../components/Network/src/WiFiEventHandler.h:18,
                 from ../components/Network/src/WiFiEventHandler.cpp:1:
/home/jeandudey/Dev/esp-idf/components/esp_event/include/esp_event_loop.h:2:2: warning: #warning "esp_event_loop.h is deprecated, please include esp_event.h instead" [-Wcpp]
 #warning "esp_event_loop.h is deprecated, please include esp_event.h instead"
  ^~~~~~~
[1185/1199] Building CXX object esp-idf/BLEPreferences/CMakeFiles/__idf_BLEPreferences.dir/src/BLEPreferences.cpp.obj
In file included from ../components/Network/src/WiFiEventHandler.h:18,
                 from ../components/Network/src/WiFi.h:22,
                 from ../components/BLEPreferences/src/BLEPreferences.cpp:19:
/home/jeandudey/Dev/esp-idf/components/esp_event/include/esp_event_loop.h:2:2: warning: #warning "esp_event_loop.h is deprecated, please include esp_event.h instead" [-Wcpp]
 #warning "esp_event_loop.h is deprecated, please include esp_event.h instead"
  ^~~~~~~
[1191/1199] Building CXX object esp-idf/WS/CMakeFiles/__idf_WS.dir/src/HttpRequest.cpp.obj
In file included from ../components/WS/src/HttpRequest.cpp:44:
../components/Util/src/Base64.h:40:20: warning: 'std::size_t util::base64EncodedLength(const string&)' defined but not used [-Wunused-function]
 static std::size_t base64EncodedLength(const std::string& in)
                    ^~~~~~~~~~~~~~~~~~~
[1194/1199] Building CXX object esp-idf/main/CMakeFiles/__idf_main.dir/main.cpp.obj
In file included from ../components/Network/src/WiFiEventHandler.h:18,
                 from ../components/Network/src/WiFi.h:22,
                 from ../main/main.cpp:23:
/home/jeandudey/Dev/esp-idf/components/esp_event/include/esp_event_loop.h:2:2: warning: #warning "esp_event_loop.h is deprecated, please include esp_event.h instead" [-Wcpp]
 #warning "esp_event_loop.h is deprecated, please include esp_event.h instead"
  ^~~~~~~
```

</details>

**To Reproduce**
1. Compile the source code

**Expected behavior**
1. No warnings

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04

--- a/components/HTTP/src/HttpServerHandler.cpp
+++ b/components/HTTP/src/HttpServerHandler.cpp
@@ -292,7 +292,6 @@ void HttpServerHandler::setUpCredentialHandler(HttpRequest* pHttpRequest, HttpRe
     std::istringstream ss(body);
     std::string token[4]; //length of json object, hoe many field there are
     std::map<std::string, std::string> json_map;
-    esp_err_t err;
 
     std::string user;
     std::string pass;

--- a/components/Network/src/WiFi.cpp
+++ b/components/Network/src/WiFi.cpp
@@ -1,10 +1,10 @@
 /**
  * @file WiFi.cpp
  * @author Locha Mesh project developers (locha.io)
- * @brief 
+ * @brief
  * @version 0.1.1
  * @date 2019-08-15
- * 
+ *
  * @copyright Copyright (c) 2019 Locha Mesh project developers
  * @license Apache 2.0, see LICENSE file for details
  */
@@ -14,7 +14,6 @@
 #include <cstdint>
 #include <cstring>
 
-#include <esp_event_loop.h>
 #include <esp_log.h>
 #include <esp_wifi.h>
 
@@ -46,10 +45,10 @@ static const std::uint8_t VALID_CHARACTERS[0xFF] = {
 
 /**
  * @brief Copy bytes from a `const char*`
- * 
+ *
  * @param dest: destination buffer, should be large enough to hold `max` bytes.
  * @param src: source buffer
- * 
+ *
  * @param max: destination maximum length
  */
 void copy_bytes(std::uint8_t* dest, const char* src, std::size_t max)
@@ -110,16 +109,32 @@ esp_err_t WiFi::init()
     esp_err_t err;
 
     ESP_LOGD(TAG, "Init TCP/IP adapter");
-    tcpip_adapter_init();
+    err = esp_netif_init();
+    if (err != ESP_OK) {
+        ESP_LOGE(TAG, "esp_netif_init failed, err = %s", esp_err_to_name(err));
+        return err;
+    }
+
+    err = esp_event_loop_create_default();
+    if (err != ESP_OK) {
+        ESP_LOGE(TAG, "Couldn't create event loop, err = %s", esp_err_to_name(err));
+        return err;
+    }
+    esp_netif_create_default_wifi_ap();
 
     ESP_LOGD(TAG, "Initializing Wi-Fi");
-
-    esp_event_loop_init(&WiFi::eventHandler, this);
 
     wifi_init_config_t init_config = WIFI_INIT_CONFIG_DEFAULT();
     err = esp_wifi_init(&init_config);
     if (err != ESP_OK) {
         ESP_LOGE(TAG, "esp_wifi_init failed, err = %s", esp_err_to_name(err));
+        return err;
+    }
+
+    err = esp_event_handler_register(WIFI_EVENT, ESP_EVENT_ANY_ID,
+                                     &WiFi::eventHandler, this);
+    if (err != ESP_OK) {
+        ESP_LOGE(TAG, "Couldn't register Wi-Fi event handler, err = %s", esp_err_to_name(err));
         return err;
     }
 
@@ -284,25 +299,26 @@ esp_err_t WiFi::stop()
     return ESP_OK;
 }
 
-esp_err_t WiFi::eventHandler(void* ctx, system_event_t* event)
+void WiFi::eventHandler(void *event_handler_arg,
+                        esp_event_base_t event_base,
+                        std::int32_t event_id,
+                        void *event_data)
 {
-    ESP_LOGD(TAG, "Wi-Fi Event Handler Called");
-    WiFi* wifi = reinterpret_cast<WiFi*>(ctx);
 
-    esp_err_t err = wifi->m_event_handler.eventDispatcher(event);
+    ESP_LOGD(TAG, "Wi-Fi Event Handler Called");
+    WiFi* wifi = reinterpret_cast<WiFi*>(event_handler_arg);
+
+    esp_err_t err = wifi->m_event_handler.eventDispatcher(event_id, event_data);
     if (err != ESP_OK) {
         ESP_LOGE(TAG, "Event handler error, err = %s", esp_err_to_name(err));
-        return err;
     }
-
-    return ESP_OK;
 }
 
 /**
  * @brief Checks if a character is a valid SSID character
- * 
+ *
  * @param c: the character
- * 
+ *
  * @return true: is valid
  * @return false: not valid
  */

--- a/components/Network/src/WiFi.h
+++ b/components/Network/src/WiFi.h
@@ -277,11 +277,17 @@ private:
     /**
      * @brief Wi-Fi event handler
      * 
-     * @param ctx: application specific data (currently a pointer to the class
-     * instance)
-     * @param event: Wi-Fi event
+     * @param[in] event_handler_arg callback handler argument (in this case
+     *                              our object).
+     * @param[in] event_base        WIFI_EVENT
+     * @param[in] event_id          Id of the event.
+     * @param[in] event_data        Data of the event (event_id specifies the
+     *                              type to use).
      */
-    static esp_err_t eventHandler(void* ctx, system_event_t* event);
+    static void eventHandler(void *event_handler_arg,
+                             esp_event_base_t event_base,
+                             std::int32_t event_id,
+                             void* event_data);
 
     WiFiDefaultEventHandler m_event_handler; /*!< Pointer to desired handler event */
 };

--- a/components/Network/src/WiFiEventHandler.cpp
+++ b/components/Network/src/WiFiEventHandler.cpp
@@ -1,13 +1,21 @@
+/**
+ * @file WiFi.h
+ * @author Locha Mesh project developers (locha.io)
+ * @brief
+ * @version 0.1.1
+ * @date 2019-08-15
+ *
+ * @copyright Copyright (c) 2019 Locha Mesh project developers
+ * @license Apache 2.0, see LICENSE file for details
+ */
+
 #include "WiFiEventHandler.h"
-#include "sdkconfig.h"
+
 #include <esp_err.h>
-#include <esp_event.h>
-#include <esp_event_loop.h>
 #include <esp_log.h>
 #include <esp_wifi.h>
 
 static const char* TAG = "WiFiEventHandler";
-
 
 namespace network {
 
@@ -16,7 +24,7 @@ WiFiEventHandler::WiFiEventHandler()
 {
 }
 
-esp_err_t WiFiEventHandler::staScanDone(system_event_sta_scan_done_t info)
+esp_err_t WiFiEventHandler::staScanDone(wifi_event_sta_scan_done_t* info)
 {
     ESP_LOGD(TAG, "default staScanDone");
     return ESP_OK;
@@ -34,25 +42,25 @@ esp_err_t WiFiEventHandler::staStop()
     return ESP_OK;
 }
 
-esp_err_t WiFiEventHandler::staConnected(system_event_sta_connected_t info)
+esp_err_t WiFiEventHandler::staConnected(wifi_event_sta_connected_t* info)
 {
     ESP_LOGD(TAG, "default staConnected");
     return ESP_OK;
 }
 
-esp_err_t WiFiEventHandler::staDisconnected(system_event_sta_disconnected_t info)
+esp_err_t WiFiEventHandler::staDisconnected(wifi_event_sta_disconnected_t* info)
 {
     ESP_LOGD(TAG, "default staDisconnected");
     return ESP_OK;
 }
 
-esp_err_t WiFiEventHandler::staAuthChange(system_event_sta_authmode_change_t info)
+esp_err_t WiFiEventHandler::staAuthChange(wifi_event_sta_authmode_change_t* info)
 {
     ESP_LOGD(TAG, "default staAuthChange");
     return ESP_OK;
 }
 
-esp_err_t WiFiEventHandler::staGotIp(system_event_sta_got_ip_t info)
+esp_err_t WiFiEventHandler::staGotIp(ip_event_got_ip_t* info)
 {
     ESP_LOGD(TAG, "default staGotIp");
     return ESP_OK;
@@ -69,75 +77,66 @@ esp_err_t WiFiEventHandler::apStop()
     ESP_LOGD(TAG, "default apStop");
     return ESP_OK;
 }
-esp_err_t WiFiEventHandler::apStaConnected(system_event_ap_staconnected_t info)
+esp_err_t WiFiEventHandler::apStaConnected(wifi_event_ap_staconnected_t* info)
 {
     ESP_LOGD(TAG, "default apStaConnected");
     return ESP_OK;
 }
 
-esp_err_t WiFiEventHandler::apStaDisconnected(system_event_ap_stadisconnected_t info)
+esp_err_t WiFiEventHandler::apStaDisconnected(wifi_event_ap_stadisconnected_t* info)
 {
     ESP_LOGD(TAG, "default apStaDisconnected");
     return ESP_OK;
 }
 
-esp_err_t WiFiEventHandler::eventDispatcher(system_event_t* event)
+esp_err_t WiFiEventHandler::eventDispatcher(std::int32_t event_id,
+                                            void* event_data)
 {
-    ESP_LOGD(TAG, "eventHandler called, event = 0x%x", event->event_id);
+    ESP_LOGD(TAG, "eventHandler called, event = 0x%x", event_id);
 
-    switch (event->event_id) {
-    case SYSTEM_EVENT_SCAN_DONE: {
-        return staScanDone(event->event_info.scan_done);
-        break;
-    }
-    case SYSTEM_EVENT_STA_START: {
+    // Call the necessary event handler
+    switch (event_id) {
+    case WIFI_EVENT_SCAN_DONE:
+        return staScanDone(reinterpret_cast<wifi_event_sta_scan_done_t*>(event_data));
+
+    case WIFI_EVENT_STA_START:
         return staStart();
-        break;
-    }
-    case SYSTEM_EVENT_STA_STOP: {
+
+    case WIFI_EVENT_STA_STOP:
         return staStop();
-        break;
-    }
-    case SYSTEM_EVENT_STA_CONNECTED: {
-        return staConnected(event->event_info.connected);
-        break;
-    }
-    case SYSTEM_EVENT_STA_DISCONNECTED: {
-        return staDisconnected(event->event_info.disconnected);
-        break;
-    }
-    case SYSTEM_EVENT_STA_AUTHMODE_CHANGE: {
-        return staAuthChange(event->event_info.auth_change);
-        break;
-    }
-    case SYSTEM_EVENT_STA_GOT_IP: {
-        return staGotIp(event->event_info.got_ip);
-        break;
-    }
-    case SYSTEM_EVENT_AP_START: {
+
+    case WIFI_EVENT_STA_CONNECTED:
+        return staConnected(reinterpret_cast<wifi_event_sta_connected_t*>(event_data));
+
+    case WIFI_EVENT_STA_DISCONNECTED:
+        return staDisconnected(reinterpret_cast<wifi_event_sta_disconnected_t*>(event_data));
+
+    case WIFI_EVENT_STA_AUTHMODE_CHANGE:
+        return staAuthChange(reinterpret_cast<wifi_event_sta_authmode_change_t*>(event_data));
+
+    case IP_EVENT_STA_GOT_IP:
+        return staGotIp(reinterpret_cast<ip_event_got_ip_t*>(event_data));
+
+    case WIFI_EVENT_AP_START:
         return apStart();
-        break;
-    }
-    case SYSTEM_EVENT_AP_STOP: {
+
+    case WIFI_EVENT_AP_STOP:
         return apStop();
-        break;
-    }
-    case SYSTEM_EVENT_AP_STACONNECTED: {
-        return apStaConnected(event->event_info.sta_connected);
-        break;
-    }
-    case SYSTEM_EVENT_AP_STADISCONNECTED: {
-        return apStaDisconnected(event->event_info.sta_disconnected);
-        break;
-    }
+
+    case WIFI_EVENT_AP_STACONNECTED:
+        return apStaConnected(reinterpret_cast<wifi_event_ap_staconnected_t*>(event_data));
+
+    case WIFI_EVENT_AP_STADISCONNECTED:
+        return apStaDisconnected(reinterpret_cast<wifi_event_ap_stadisconnected_t*>(event_data));
+
     default:
-        ESP_LOGD(TAG, "unknown event = %d", event->event_id);
+        ESP_LOGD(TAG, "unknown event = %d", event_id);
         break;
     }
 
     if (m_next_handler) {
         ESP_LOGD(TAG, "Dispatching event to the next_handler");
-        return m_next_handler->eventDispatcher(event);
+        return m_next_handler->eventDispatcher(event_id, event_data);
     }
 
     return ESP_OK;

--- a/components/Network/src/WiFiEventHandler.h
+++ b/components/Network/src/WiFiEventHandler.h
@@ -1,10 +1,10 @@
 /**
  * @file WiFiEventHandler.h
  * @author Locha Mesh project developers (locha.io)
- * @brief 
+ * @brief
  * @version 0.1
  * @date 2019-12-02
- * 
+ *
  * @copyright Copyright (c) 2019 Locha Mesh project developers
  * @license Apache 2.0, see LICENSE file for details
  */
@@ -15,7 +15,8 @@
 #include <memory>
 
 #include <esp_event.h>
-#include <esp_event_loop.h>
+#include <esp_wifi.h>
+#include <esp_netif.h>
 
 namespace network {
 
@@ -24,7 +25,7 @@ namespace network {
  *
  * Typically this class is subclassed to provide implementations for the
  * callbacks we want to handle:
- * 
+ *
  * @code{cpp}
  * class MyHandler : public WiFiEventHandler {
  *     esp_err_t apStart() {
@@ -39,7 +40,7 @@ class WiFiEventHandler
 public:
     /**
      * @brief Construct a new Wi-Fi event handler object
-     * 
+     *
      */
     WiFiEventHandler();
 
@@ -47,16 +48,16 @@ public:
      * @brief "Scan Done" handler
      *
      * @param info: event information
-     * 
+     *
      * @return
      *      - ESP_OK: succeed
      *      - (others): failed
      */
-    virtual esp_err_t staScanDone(system_event_sta_scan_done_t info);
+    virtual esp_err_t staScanDone(wifi_event_sta_scan_done_t* info);
 
     /**
      * @brief "Station Start" handler
-     * 
+     *
      * @return
      *      - ESP_OK: succeed
      *      - (others): failed
@@ -65,7 +66,7 @@ public:
 
     /**
      * @brief "Station Stop" handler
-     * 
+     *
      * @return
      *      - ESP_OK: succeed
      *      - (others): failed
@@ -74,51 +75,51 @@ public:
 
     /**
      * @brief "Station Connected" handler
-     * 
+     *
      * @param info: event information
-     * 
+     *
      * @return
      *      - ESP_OK: succeed
      *      - (others): failed
      */
-    virtual esp_err_t staConnected(system_event_sta_connected_t info);
+    virtual esp_err_t staConnected(wifi_event_sta_connected_t* info);
 
     /**
-     * @brief 
-     * 
+     * @brief "Station Disconnected" event
+     *
      * @param info: event information
-     * 
+     *
      * @return
      *      - ESP_OK: succeed
      *      - (others): failed
      */
-    virtual esp_err_t staDisconnected(system_event_sta_disconnected_t info);
+    virtual esp_err_t staDisconnected(wifi_event_sta_disconnected_t* info);
 
     /**
      * @brief "Authentication Mode Change" handler
-     * 
+     *
      * @param info: event information
-     * 
+     *
      * @return
      *      - ESP_OK: succeed
      *      - (others): failed
      */
-    virtual esp_err_t staAuthChange(system_event_sta_authmode_change_t info);
+    virtual esp_err_t staAuthChange(wifi_event_sta_authmode_change_t* info);
 
     /**
      * @brief "Station Got IP" handler
-     * 
+     *
      * @param info: event information
-     * 
+     *
      * @return
      *      - ESP_OK: succeed
      *      - (others): failed
      */
-    virtual esp_err_t staGotIp(system_event_sta_got_ip_t info);
+    virtual esp_err_t staGotIp(ip_event_got_ip_t* info);
 
     /**
      * @brief "AP Start" handler
-     * 
+     *
      * @return
      *      - ESP_OK: succeed
      *      - (others): failed
@@ -127,7 +128,7 @@ public:
 
     /**
      * @brief "AP Stop" handler
-     * 
+     *
      * @return
      *      - ESP_OK: succeed
      *      - (others): failed
@@ -136,31 +137,31 @@ public:
 
     /**
      * @brief "AP Station Connected" handler
-     * 
+     *
      * @param info: event information
-     * 
+     *
      * @return
      *      - ESP_OK: succeed
      *      - (others): failed
      */
-    virtual esp_err_t apStaConnected(system_event_ap_staconnected_t info);
+    virtual esp_err_t apStaConnected(wifi_event_ap_staconnected_t* info);
 
     /**
      * @brief "Station Disconnected" from AP event
-     * 
+     *
      * @param info: event information
-     * 
+     *
      * @return
      *      - ESP_OK: succeed
      *      - (others): failed
      */
-    virtual esp_err_t apStaDisconnected(system_event_ap_stadisconnected_t info);
+    virtual esp_err_t apStaDisconnected(wifi_event_ap_stadisconnected_t* info);
 
     /**
-	 * @brief Set the next WiFi event handler in the chain
-     * 
-	 * @param next_handler: the next Wi-Fi event handler in the chain
-	 */
+     * @brief Set the next WiFi event handler in the chain
+     *
+     * @param next_handler: the next Wi-Fi event handler in the chain
+     */
     void setNextHandler(std::unique_ptr<WiFiEventHandler>&& next_handler)
     {
         m_next_handler = std::move(next_handler);
@@ -168,14 +169,15 @@ public:
 
     /**
      * @brief Dispatch events by calling the respective WiFiEventHander methods
-     * 
-     * @param event: the Wi-Fi event
-     * 
+     *
+     * @param[in] event_id   The Wi-Fi event ID.
+     * @param[in] event_data Data about the event
+     *
      * @return
      *      - ESP_OK: succeed
      *      - (others): failed
      */
-    esp_err_t eventDispatcher(system_event_t* event);
+    esp_err_t eventDispatcher(std::int32_t event_id, void* event_data);
 
 private:
     std::unique_ptr<WiFiEventHandler> m_next_handler;

